server/peers.py

```diff
@@ -560,9 +560,11 @@ def retry_peer(self, peer, port_pairs):
             create_connection = self.proxy.create_connection
         else:
             create_connection = self.loop.create_connection
+            
+        laddr=(self.env.host, None) if self.env.host else None    
```

Comment:
Do you not need to set the port in the tuple?  I can't test as I don't have this setup.